includes/process-download.php

```diff
@@ -1076,6 +1098,53 @@ function edd_local_file_location_is_allowed( $file_details, $schemas, $requested
 	return apply_filters( 'edd_local_file_location_is_allowed', $should_allow, $file_details, $schemas, $requested_file );
 }
 
+/**
+ * Detect downloading a file immeidatly after a forced login.
+ *
+ * When the store requires being logged in to download files, this handles the file download after logging in.
+ * We need this otherwise the file is downloaded immediatly after successfully logging in, but the page never changes.
+ *
+ * @since 3.1
+ */
+function edd_redirect_file_download_after_login() {
+	$nonce = isset( $_GET['_nonce'] ) ? sanitize_text_field( $_GET['_nonce'] ) : false;
+
+	// No nonce provided, redirect to the homepage.
+	if ( empty( $nonce ) ) {
+		wp_safe_redirect( home_url() );
+	}
+
+	// Nonce verification failed, redirect to the homepage.
+	if ( wp_verify_nonce( $nonce, 'edd_process_file_download_after_login' ) ) {
```

Comment:
```suggestion
	if ( ! wp_verify_nonce( $nonce, 'edd_process_file_download_after_login' ) ) {
```
I think this is inverted? Although changing it is not allowing a download (either way I don't get my file after a successful login and I'm not seeing an error message for why).

Reply:
Ah yep I was testing some things and committed my test.